pkg/vere/io/http.c

```diff
@@ -650,20 +650,38 @@ _http_req_dispatch(u3_hreq* req_u, u3_noun req)
 /* _http_cache_respond(): respond with a simple-payload:http
 */
 static void
-_http_cache_respond(u3_hreq* req_u, u3_noun nun) {
-  // XX check auth
-  u3_noun auth, response_header, data;
-  u3x_qual(u3k(u3t(u3t(nun))), &auth, 0, &response_header, &data);
-  u3_noun status, headers;
-  u3x_cell(response_header, &status, &headers);
+_http_cache_respond(u3_hreq* req_u, u3_noun nun, u3_noun req) {
+  h2o_req_t* rec_u = req_u->rec_u;
+  u3_httd* htd_u = req_u->hon_u->htp_u->htd_u;
 
-  req_u->sat_e = u3_rsat_plan;
-  _http_start_respond(req_u, u3k(status), u3k(headers), u3k(data), c3y);
+  if ( u3_none == nun ) {
+    h2o_send_error_500(rec_u, "Internal Server Error", "scry failed", 0);
+  }
+  else if ( (u3_none == u3t(nun)) || (u3_none == u3t(u3t(nun))) ) {
+    h2o_send_error_404(rec_u, "Not Found", "not found", 0);
+  }
```

Comment:
These checks won't work: `u3t()` will bail on an atom, not return `u3_none` (which is "not a known", distinct from `u3_nul`).

Because we used a "last" scry, `nun` is `(unit ,[%omen path (cask)])`. The arvo/vere interface collapses the outer unit from the vane scry (`~` and `[~ ~]` both become `~`).  And `%eyre`'s `/cache` scry interface doesn't let us distinguish between failed scrys and empty cache entries.

I think we want `u3_nul == nun` => 404, and everything else 500. We can't match %eyre's request handling behavior exactly wrt 404s, but they're probably preferable regardless.

You can use `u3r_at(7, nun)` to get a `u3_weak` equivalent to `u3t(u3t(nun))` (check against `u3_none`), or you could check `c3n == u3r_trel(nun, 0, 0, &some_noun)`.

Reply:
fixed, thanks!